target/linux/siflower/dts/sf21a6826p.dtsi

```diff
@@ -0,0 +1,921 @@
+			reg = <2>;
+			i-cache-block-size = <64>;
+			i-cache-size = <32768>;
+			i-cache-sets = <128>;
+			d-cache-block-size = <64>;
+			d-cache-size = <32768>;
+			d-cache-sets = <128>;
+			riscv,cbom-block-size = <64>;
+			riscv,cbop-block-size = <64>;
+			riscv,cboz-block-size = <64>;
+			clocks = <&realcpuclk 0>;
+			next-level-cache = <&l2_cache>;
+			mmu-type = "riscv,sv39";
+
+			cpu2_intc: interrupt-controller {
+				#interrupt-cells = <1>;
+				compatible = "riscv,cpu-intc";
+				interrupt-controller;
+			};
+		};
+
+		cpu@3 {
+			compatible = "riscv";
+			device_type = "cpu";
+			riscv,isa = "rv64imafdc";
+			riscv,isa-base = "rv64i";
+			riscv,isa-extensions = "i", "m", "a", "f", "d", "c",
+						"sstc", "svinval", "svnapot", "svpbmt", "zba", "zbb", "zbc", "zbs", "zicbom", "zicbop", "zicboz", "zicntr", "zicsr", "zifencei", "zihintpause", "zihpm";
+			reg = <3>;
+			i-cache-block-size = <64>;
+			i-cache-size = <32768>;
+			i-cache-sets = <128>;
+			d-cache-block-size = <64>;
+			d-cache-size = <32768>;
+			d-cache-sets = <128>;
+			riscv,cbom-block-size = <64>;
+			riscv,cbop-block-size = <64>;
+			riscv,cboz-block-size = <64>;
+			clocks = <&realcpuclk 0>;
+			next-level-cache = <&l2_cache>;
+			mmu-type = "riscv,sv39";
+
+			cpu3_intc: interrupt-controller {
+				#interrupt-cells = <1>;
+				compatible = "riscv,cpu-intc";
+				interrupt-controller;
+			};
+		};
+
+		l2_cache: l2-cache {
+			compatible = "cache";
+			cache-level = <2>;
+			cache-block-size = <64>;
+			cache-size = <262144>;
+			cache-sets = <256>;
+			cache-unified;
+		};
+	};
+
+	clocks {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		osc:oscillator {
+			compatible = "fixed-clock";
+			#clock-cells = <0>;
+			clock-frequency = <25000000>;
+			clock-names = "osc_clk";
+			clock-output-names = "osc";
+		};
+		cmnfoutpostdiv:cmnfoutpostdiv {
+			compatible = "siflower,sf21a6826p-cmn-pll";
+			#clock-cells = <1>;
+			clocks = <&osc>;
+			clock-output-names = "cmnfoutpostdiv";
+			div-shift = <6>;
+			reg = <0x0CE00400 0x4>;
+		};
+		ddrfoutpostdiv:ddrfoutpostdiv {
+			compatible = "siflower,sf21a6826p-ddr-pll";
+			#clock-cells = <1>;
+			clocks = <&osc>;
+			clock-output-names = "ddrfoutpostdiv";
+			div-shift = <24>;
+			reg = <0x0CE00418 0xC>;
+		};
+		pciefout1:pciefout1 {
+			compatible = "siflower,sf21a6826p-pcie-pll";
+			#clock-cells = <1>;
+			clocks = <&osc>;
+			clock-output-names = "pciefout1";
+			div-shift = <12>;
+			reg = <0x0CE0044C 0xC>;
+		};
+		pciefout2:pciefout2 {
+			compatible = "siflower,sf21a6826p-pcie-pll";
+			#clock-cells = <1>;
+			clocks = <&osc>;
+			clock-output-names = "pciefout2";
+			div-shift = <6>;
+			reg = <0x0CE0044C 0xC>;
+		};
+		pciefout3:pciefout3 {
+			compatible = "siflower,sf21a6826p-pcie-pll";
+			#clock-cells = <1>;
+			clocks = <&osc>;
+			clock-output-names = "pciefout3";
+			div-shift = <0>;
+			reg = <0x0CE0044C 0xC>;
+		};
+		cpuclk:cpuclk {
```

Comment:
It's probably unnecessary to split the clock driver like this. These can be merged into a single dt node providing multiple clocks, like what other platforms are doing.

Reply:
I agree that it needs an overhaul. However the maintainer of this driver has resigned, so it may take a while.